### PR TITLE
fix: Hard-coded link in Header component changed by LMS_BASE_URL env var

### PR DESCRIPTION
**Pull request by ChrisChV**
`index.jsx` file of Header component updated with change the hard-coded link of `https://www.edx.org` with a component property.
`index.jsx` file of root project updated. Added the environment variable `LMS_BASE_URL` to the parameters of the Header component

Issue: https://github.com/openedx/build-test-release-wg/issues/55

**What changed?**

- This updates the `index.jsx` file of the Header component with change the hard-coded link with the component property `logoDestination`.
- This updates the `index.jsx` file of root project with add the `LMS_BASE_URL` env var as `logoDestination` in Header component.

**Developer Checklist**
- [X] Test suites passing
- [ ] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

If you start your local development server for gradebook and you load a course, when you click in the header logo, then it should redirect to the LMS dashboard.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x]  I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ChrisChV! I've created [OSPR-5806](https://openedx.atlassian.net/browse/OSPR-5806) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.




:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://openedx.org/cla)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by sarina** [comment]
@davidjoy didn't you fix this already?

**Comment by natabene** [comment]
@ChrisChV Thank you for the contribution! Once you have signed our Contributor Agreement and it has been processed, you will receive a confirmation via email from our Legal team. We then will be able to review this and all your future code contributions.

**Comment by essiocolom** [comment]
@natabene we already signed the Entity Contributor Agreement and sent it to legal as we are going to be contributing as Bitmaker and @ChrisChV is part of the organization 👍  By the way, we are still waiting for them to confirm everything is okay

**Comment by natabene** [comment]
@ChrisChV Thank you for an update! They will process and get back to you soon.

**Comment by ChrisChV** [comment]
Hi @natabene.  We have already been authorized as a bitmaker to contribute

**Comment by davidjoy** [comment]
Looking around at the shared MFE header, I think we should set the logo destination to point at the dashboard for consistency:

`${config.LMS_BASE_URL}/dashboard`

In which case we don't need to add a new environment variable specific to gradebook. 

See shared header implementation here: https://github.com/edx/frontend-component-header/blob/3726792df260e72bcf9be6cd3bc921b880255604/src/Header.jsx#L82

**Comment by ChrisChV** [comment]
> Looking around at the shared MFE header, I think we should set the logo destination to point at the dashboard for consistency:
> 
> `${config.LMS_BASE_URL}/dashboard`
> 
> In which case we don't need to add a new environment variable specific to gradebook.
> 
> See shared header implementation here: https://github.com/edx/frontend-component-header/blob/3726792df260e72bcf9be6cd3bc921b880255604/src/Header.jsx#L82

@davidjoy In this case it is better to place the logo destination value in the Header as I have done now? Or pass that value through the Header props from the index file, as being done in the link you shared?

**Comment by natabene** [comment]
@ChrisChV Thank you for joining the Open edX community! We have added your user to our records and can now review your code going forward. @davidjoy Would you like to review?

**Comment by natabene** [comment]
@mattcarter FYI this PR coming.

**Comment by ChrisChV** [comment]
@davidjoy Please review the new changes, I added a new logic to solve this using what you commented and without creating a new env var

**Comment by natabene** [comment]
@mattcarter Do you want to have a look now?

**Comment by jansenk** [comment]
@mattcarter @natabene I've got some spare cycles, I can review this if you'd like

**Comment by mattcarter** [comment]
@jansenk, that would be great.  Please review and engage with @davidjoy

**Review by jansenk — APPROVED** [review]
Looks good! Small style suggestion

File: src/components/Header/index.jsx [hunk 1]
```diff
@@ -3,17 +3,26 @@ import { Hyperlink } from '@edx/paragon';
...
   render() {
+    const {
+      logoDestination
+    } = this.props;
+    const logoProps = { destination: logoDestination}
     return (
       <div className="mb-3">
         <header className="d-flex justify-content-center align-items-center p-3 border-bottom-blue">
-          <Hyperlink destination="https://www.edx.org">
+          <Hyperlink {...logoProps}>
```
[the remark this answers is not in the document]
Comment (davidjoy) [reply src/components/Header/index.jsx:21]: I think this line should just read:

```
<Hyperlink destination={`${getConfig().LMS_BASE_URL}/dashboard`}>
```

It's not really a configurable thing/there doesn't seem to be much value in passing it down from above.

**Review by davidjoy — CHANGES_REQUESTED** [review]
I think we can simplify this and just put the URL in the hyperlink.  The index.jsx file doesn't really need to know about the URL/this code isn't reused with different URLs.

**Comment by ChrisChV** [comment]
@davidjoy It's done

**Review by davidjoy — APPROVED** [review]

**Comment by natabene** [comment]
@davidjoy @jansenk If this is good to merge, let's merge.

**Comment by edx-webhook** [comment]
@ChrisChV 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/frontend-app-gradebook&entry.867055334=https://github.com/edx/frontend-app-gradebook/pull/186&entry.1484655318=https://github.com/ChrisChV&entry.752974735=2021-05-25+21:42&entry.1917517419=2021-06-14+17:51&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**